File: server/src/main/java/com/ishop/ishoptech/entities/Image.java
package com.ishop.ishoptech.entities;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.*;
import org.hibernate.annotations.Fetch;
import org.hibernate.annotations.FetchMode;

import javax.persistence.*;

@Data
@Table
@Entity(name = "img")
@With
@AllArgsConstructor
@NoArgsConstructor
public class Image {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long idImg;

    private String description;
    private String pathToFile;

    @ToString.Exclude
    @JsonProperty(access = JsonProperty.Access.WRITE_ONLY)
    @ManyToOne(fetch = FetchType.LAZY)
    @Fetch(FetchMode.JOIN)
    @JoinColumn(name = "idProduct")
    private Product product;

}

File: client/shop/src/app/models/image/Image.ts
export class Image {
  id: number;
  description: string;
  path_to_file: string;

  constructor(id: number, description: string, path_to_file: string) {
    this.id = id;
    this.description = description;
    this.path_to_file = path_to_file;
  }


}
